gymnasium/vector/vector_env.py

```diff
@@ -235,81 +235,197 @@ class VectorWrapper(VectorEnv):
         Don't forget to call ``super().__init__(env)`` if the subclass overrides :meth:`__init__`.
     """
 
-    def __init__(self, env: VectorEnv):
+    def __init__(self, env: VectorEnv[VectorObsType, VectorActType, VectorArrayType]):
+        """Initialises the wrapper with :attr:`env` to wrap and an observation and action space."""
         super().__init__()
 
         assert isinstance(env, VectorEnv)
         self.env = env
 
-    # explicitly forward the methods defined in VectorEnv
-    # to self.env (instead of the base class)
+        self._action_space: spaces.Space[ActType] | None = None
+        self._observation_space: spaces.Space[ObsType] | None = None
+        self._metadata: dict[str, Any] | None = None
 
-    def reset(self, **kwargs):
-        return self.env.reset(**kwargs)
+    def reset(
+        self,
+        *,
+        seed: int | list[int] | None = None,
+        options: dict[str, Any] | None = None,
+    ) -> tuple[VectorWrapperObsType, dict[str, Any]]:
+        """Calls :meth:`reset` on the :attr:`env` allowing for modification of observations, info or options."""
+        return self.env.reset(seed=seed, options=options)
 
-    def step(self, actions):
+    def step(
+        self, actions: VectorActType
+    ) -> tuple[
+        VectorWrapperObsType,
+        VectorWrapperArrayType,
+        VectorWrapperArrayType,
+        VectorWrapperArrayType,
+        dict[str, Any],
+    ]:
+        """Calls :meth:`step` on the :attr:`env` allowing for modification of actions, observations, etc."""
         return self.env.step(actions)
 
-    def close(self, **kwargs):
-        return self.env.close(**kwargs)
+    def render(self) -> RenderFrame | list[RenderFrame] | None:
+        """Uses the :meth:`render` of the :attr:`env` that can be overwritten to change the returned data."""
+        return self.env.render()
 
-    def close_extras(self, **kwargs):
-        return self.env.close_extras(**kwargs)
+    def close(self, **kwargs):
```

Comment:
(Note: this will have to be explicitly noted in the changelog)

Reply:
Agreed